### PR TITLE
[indexer] Create petra_activities table.

### PR DESCRIPTION
**Pull request by zacharydenton**
### Description

Adds a new table to indexer's token_processor that contains coin_activities and token_activities grouped as one row per transaction per account affected by the transaction.

```
 transaction_version |                          account_address                           |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 coin_activities                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | token_activities 
---------------------+--------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------
             4980001 | 0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4 | [{"amount": "54100", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": -1, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": 3, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "event_creation_number": 2, "event_sequence_number": 18, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | []
             4980001 | 0x0619dc29a0aac8fa146714058e8dd6d2d0f3bdf5f6331907bf91f3acd81e6935 | [{"amount": "54100", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": -1, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": 3, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "event_creation_number": 2, "event_sequence_number": 18, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | []
             4980001 | 0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6 | [{"amount": "54100", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": -1, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": 3, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "2440000", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xbf3cb724ea6eae637284c0a3ac0937c8e961c9720505372ec5903d7a9ad016c4", "event_creation_number": 2, "event_sequence_number": 18, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | []
             4980002 | 0x0619dc29a0aac8fa146714058e8dd6d2d0f3bdf5f6331907bf91f3acd81e6935 | [{"amount": "321700", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": -1, "event_sequence_number": 20, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "1358460", "coin_type": "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 16, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "16423495", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 3, "event_sequence_number": 12, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "0", "coin_type": "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 15, "event_sequence_number": 1, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "79781", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 2, "event_sequence_number": 8, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "4463961", "coin_type": "0x5a97986a9d031c4567e15b797be516910cfcb4156312482efc6a19c0a30c948::lp_coin::LP<0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT, 0x1::aptos_coin::AptosCoin, 0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::curves::Uncorrelated>", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 17, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}] | []
             4980002 | 0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515 | [{"amount": "321700", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": -1, "event_sequence_number": 20, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "1358460", "coin_type": "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 16, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "16423495", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::WithdrawEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 3, "event_sequence_number": 12, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "0", "coin_type": "0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 15, "event_sequence_number": 1, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "79781", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 2, "event_sequence_number": 8, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}, {"amount": "4463961", "coin_type": "0x5a97986a9d031c4567e15b797be516910cfcb4156312482efc6a19c0a30c948::lp_coin::LP<0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa::asset::USDT, 0x1::aptos_coin::AptosCoin, 0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::curves::Uncorrelated>", "is_gas_fee": false, "block_height": 1499010, "activity_type": "0x1::coin::DepositEvent", "owner_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "transaction_version": 4980002, "entry_function_id_str": "0x190d44266241744264b964a37b8f09863167a12d3e70cda39376cfb4e3561e12::scripts::add_liquidity", "event_account_address": "0x4b400707d50d4c3de88db87d9018b68f41b654c73dd561121f25c5205af1b515", "event_creation_number": 17, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}] | []
(5 rows)
```

```
aptos_indexer_mainnet=# explain analyze select * from petra_activities where account_address = '0x5e19c74fc3ffe6fc24dc9637db979586cb68a7568e6073ca31b8684e3b406ba7' order by transaction_version desc;
                                                                      QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------
 Index Scan Backward using petra_activities_pkey on petra_activities  (cost=0.56..348.48 rows=86 width=1636) (actual time=0.025..0.028 rows=3 loops=1)
   Index Cond: ((account_address)::text = '0x5e19c74fc3ffe6fc24dc9637db979586cb68a7568e6073ca31b8684e3b406ba7'::text)
 Planning Time: 0.081 ms
 Execution Time: 0.040 ms
(4 rows)
```

### Test Plan
I cherry-picked the PR into my local mainnet branch and ran the indexer against mainnet.

### CODE REVIEW

File: crates/indexer/src/models/token_models/petra_activities.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            let coin_activities_json =
+                serde_json::to_value(coin_activities).unwrap_or(EMPTY_JSON_ARRAY);
+            let token_activities_json =
+                serde_json::to_value(token_activities).unwrap_or(EMPTY_JSON_ARRAY);
+
+            // Add the transaction sender.
+            addresses.insert(user_txn.request.sender.to_string());
+
+            // Add any other accounts referenced in the writeset.
+            for wsc in user_txn.info.changes.iter() {
```
Comment (bowenyang007) [inline crates/indexer/src/models/token_models/petra_activities.rs:45]: i don't think you need this given all the content are coming from events? You should be able to get this from reusing the TokenActivity and CoinActivity objects.

File: crates/indexer/src/processors/token_processor.rs [hunk 2]
```diff
@@ -486,11 +515,17 @@ impl TransactionProcessor for TokenTransactionProcessor {
         let table_handle_to_owner =
             TableMetadataForToken::get_table_handle_to_owner_from_transactions(&transactions);
 
+        // get aptos_coin info for supply tracking
+        // TODO: This only needs to be fetched once. Need to persist somehow
+        let maybe_aptos_coin_info =
+            &CoinInfoQuery::get_by_coin_type(APTOS_COIN_TYPE.to_string(), &mut conn).unwrap();
```
Comment (bowenyang007) [inline crates/indexer/src/processors/token_processor.rs:521]: Is this permanent? I think you should be able to reuse the coin stuff no?

**Review by bowenyang007 — COMMENTED** [review]
early comments, overall makes sense :)

File: crates/indexer/src/processors/token_processor.rs (hunk 2 above)
Comment (zacharydenton) [reply crates/indexer/src/processors/token_processor.rs:521]: It is just copied from the way coin_processor computes the CoinActivity.

https://github.com/aptos-labs/aptos-core/blob/12491736878d172d730cfe4cda9e8e50239e6328/crates/indexer/src/processors/coin_processor.rs#L256-L259

https://github.com/aptos-labs/aptos-core/blob/12491736878d172d730cfe4cda9e8e50239e6328/crates/indexer/src/processors/coin_processor.rs#L269-L275

File: crates/indexer/src/models/token_models/petra_activities.rs (hunk 1 above)
Comment (zacharydenton) [reply crates/indexer/src/models/token_models/petra_activities.rs:45]: Good point, will update.

**Comment by zacharydenton** [comment]
@kent-white @bowenyang007 I'm debating whether to add `coin_infos` (hash of coin type => coin_info for all coin_activities in the row) and `token_datas` (hash of token_data_id_hash => token_data for all token_activities in the row), or whether the client/upstream service should simply make additional queries for that information.

**Comment by bowenyang007** [comment]
> {"amount": "54100", "coin_type": "0x1::aptos_coin::AptosCoin", "is_gas_fee": true, "block_height": 1499010, "activity_type": "0x1::aptos_coin::GasFeeEvent", "owner_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "transaction_version": 4980001, "entry_function_id_str": "0x1::coin::transfer", "event_account_address": "0xa7c0fb9acaae1208b141f6b94f768e0daa14c4722b09074816925355d73875c6", "event_creation_number": -1, "event_sequence_number": 0, "transaction_timestamp": "2022-10-20T04:01:35", "is_transaction_success": true}

I don't think that you need block_height, owner_address, transaction_version, event_account_address, entry_function_id_str, event_creation_number, event_sequence_number, transaction_timestamp, and is_transaction_success. Some of these should be at the row level and some of these are not needed. Since you have an array the ordering is already accounted for.

**Comment by bowenyang007** [comment]
I was also talking w/ @kent-white about the token data uri. You have to follow a similar design as how we get collection creator https://github.com/aptos-labs/aptos-core/blob/main/crates/indexer/src/models/token_models/collection_datas.rs#L159-L161

**Review by bowenyang007 — CHANGES_REQUESTED** [review]
Pushing back to your queue.

**Comment by kent-white** [comment]
> I'm debating whether to add coin_infos (hash of coin type => coin_info for all coin_activities in the row) and token_datas (hash of token_data_id_hash => token_data for all token_activities in the row), or whether the client/upstream service should simply make additional queries for that information.

@zacharydenton Wouldn't that take a lot of queries per row?

**Comment by davidiw** [comment]
stale?

**Comment by zacharydenton** [comment]
Yes, we're moving forward with a different approach that doesn't involve adding a new table to the indexer db.

**Pull request closed without merging**